api/rbac/rbac.go

```diff
@@ -0,0 +1,191 @@
+
+	configmapadapter "github.com/percona/everest/api/rbac/configmap-adapter"
+	"github.com/percona/everest/data"
+	"github.com/percona/everest/pkg/common"
+	"github.com/percona/everest/pkg/kubernetes"
+	"github.com/percona/everest/pkg/kubernetes/informer"
+	"github.com/percona/everest/pkg/session"
+)
+
+// Setup a new informer that watches our RBAC ConfigMap.
+// This informer reloads the policy whenever the ConfigMap is updated.
+func refreshEnforcerInBackground(
+	ctx context.Context,
+	kubeClient *kubernetes.Kubernetes,
+	enforcer *casbin.Enforcer,
+	l *zap.SugaredLogger,
+) error {
+	inf, err := informer.New(
+		informer.WithConfig(kubeClient.Config()),
+		informer.WithLogger(l),
+		informer.Watches(&corev1.ConfigMap{}, kubeClient.Namespace()),
+	)
+	inf.OnUpdate(func(_, newObj interface{}) {
+		cm, ok := newObj.(*corev1.ConfigMap)
+		if !ok || cm.GetName() != common.EverestRBACConfigMapName {
+			return
+		}
+		if err := enforcer.LoadPolicy(); err != nil {
+			l.Error("failed to load policy", zap.Error(err))
+		}
+	})
+	if inf.Start(ctx, &corev1.ConfigMap{}) != nil {
+		return errors.Join(err, errors.New("failed to watch RBAC ConfigMap"))
+	}
+	return nil
+}
+
+// NewEnforcer creates a new Casbin enforcer with the RBAC model and ConfigMap adapter.
+func NewEnforcer(ctx context.Context, kubeClient *kubernetes.Kubernetes, l *zap.SugaredLogger) (*casbin.Enforcer, error) {
+	modelData, err := fs.ReadFile(data.RBAC, "rbac/model.conf")
+	if err != nil {
+		return nil, errors.Join(err, errors.New("could not read casbin model"))
+	}
+
+	model, err := model.NewModelFromString(string(modelData))
+	if err != nil {
+		return nil, errors.Join(err, errors.New("could not create casbin model"))
+	}
+
+	cmReq := types.NamespacedName{
+		Namespace: kubeClient.Namespace(),
+		Name:      common.EverestRBACConfigMapName,
+	}
+	adapter := configmapadapter.NewAdapter(kubeClient, cmReq)
+
+	enforcer, err := casbin.NewEnforcer(model, adapter, true)
+	if err != nil {
+		return nil, errors.Join(err, errors.New("could not create casbin enforcer"))
+	}
+
+	return enforcer, refreshEnforcerInBackground(ctx, kubeClient, enforcer, l)
+}
+
+// UserGetter is a function that extracts the user from the JWT token.
+func UserGetter(c echo.Context) (string, error) {
+	token, ok := c.Get("user").(*jwt.Token) // by default token is stored under `user` key
+	if !ok {
+		return "", errors.New("failed to get token from context")
+	}
+
+	claims, ok := token.Claims.(jwt.MapClaims) // by default claims is of type `jwt.MapClaims`
+	if !ok {
+		return "", errors.New("failed to get claims from token")
+	}
+
+	subject, err := claims.GetSubject()
+	if err != nil {
+		return "", errors.Join(err, errors.New("failed to get subject from claims"))
+	}
+
+	issuer, err := claims.GetIssuer()
+	if err != nil {
+		return "", errors.Join(err, errors.New("failed to get issuer from claims"))
+	}
+
+	if issuer == session.SessionManagerClaimsIssuer {
+		return strings.Split(subject, ":")[0], nil
+	}
```

Comment:
It has nothing to do with your PR but this got me questioning the usefulness of the capabilities suffix...

Reply:
Hmm yeah. Even I don't know yet.. But we can revisit it later if its not useful.
Resolving for now to merge..